docs/USAGE-GUIDE.md

```diff
@@ -486,6 +486,30 @@ To use certificate transparency, the requester (usually the client) must call `s
 
 Call `s2n_connection_get_sct_list()` to retrieve the received certificate transparency information. The format of this data is the SignedCertificateTimestampList structure defined in section 3.3 of RFC 6962.
 
+## Session Resumption
+
+TLS handshake sessions are CPU-heavy due to the calculations involved in authenticating a certificate. These calculations can be skipped after the first connection by turning on session resumption. This mechanism stores state from the previous session and uses it to establish the next session, allowing the handshake to skip the costly authentication step while keeping the same cryptographic guarantees. Note that this session state is encrypted by the server, so a server will have to set up an external key in order to do session resumption.
+
+### Stateless Session Resumption
+
+In stateless session resumption the server creates a 'session ticket' from some state after a successful handshake. The server sends this ticket to a client, and the client can send that ticket back to the server during a new connection to skip the authentication step. This mechanism allows servers to avoid storing individual state for each client, and for that reason is the preferred method for resuming a session.
+
+#### A Note on TLS1.3 Session Resumption
+
```

Comment:
Should this be under the stateless section? If we added stateful for TLS1.3, you'd have the same issues.